sw/device/tests/crypto/ecdsa_p256_verify_functest.c

```diff
@@ -0,0 +1,88 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/drivers/hmac.h"
+#include "sw/device/lib/crypto/drivers/otbn.h"
+#include "sw/device/lib/crypto/ecdsa_p256/ecdsa_p256.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/tests/crypto/ecdsa_p256_verify_testvectors.h"
+
+hmac_error_t compute_digest(size_t msg_len, const uint8_t *msg,
+                            ecdsa_p256_message_digest_t *digest) {
+  // Compute the SHA-256 digest using the HMAC device.
+  hmac_sha256_init();
+  hmac_error_t err = hmac_sha256_update(msg, msg_len);
+  if (err != kHmacOk) {
+    return err;
+  }
+  hmac_digest_t hmac_digest;
+  err = hmac_sha256_final(&hmac_digest);
+  if (err != kHmacOk) {
+    return err;
+  }
+
+  // Copy digest into the destination array.
+  memcpy(digest->h, hmac_digest.digest, sizeof(hmac_digest.digest));
+
+  return kHmacOk;
+}
+
+bool ecdsa_p256_verify_test(const ecdsa_p256_verify_test_vector_t *testvec) {
+  // Hash message.
+  ecdsa_p256_message_digest_t digest;
+  hmac_error_t hash_err =
+      compute_digest(testvec->msg_len, testvec->msg, &digest);
+  if (hash_err != kHmacOk) {
```

Comment:
I'd make this a CHECK-fail tbh. CHECK takes a logging string after it as if by LOG_ERROR.

Reply:
Fixed, and added some messages to the `CHECK`s below as well!